TFNativeOp.py

```diff
@@ -243,7 +243,12 @@ def make_dim_str(c):
     for in_idx, v in enumerate(in_info):
       out_idx = v.get("want_inplace", -1)
       if out_idx >= 0:
-        code_forward_io += "context->forward_ref_input_to_ref_output(%i, %i);\n" % (in_idx, out_idx)
+        code_forward_io += """
+if (IsRefType(context->input_dtype({in_idx})))
+  context->forward_ref_input_to_ref_output({in_idx}, {out_idx});
+else
+  context->set_output({out_idx}, context->input({in_idx}));
+\n""".format(in_idx=in_idx, out_idx=out_idx)
```

Comment:
Why the `\n`? Also, indent it to the other Python code, which looks a bit nicer in the Python code (I don't care much how the resulting C++ code is indented...).

Reply:
Btw, also add a comment here. Did you investigate further what this actually means, and why it happens now, and not earlier?
@curufinwe You said in your compilation of TF 1.15, someone observed some strange behavior in training. Maybe this is related?

Reply:
I'll fix the formatting. 

The `forward_ref...` bit expects to have tensor references both on input and output, and sets `output = input`.  It seems that this assumption was not satisfied in some tests, i.e. the input was actually a tensor value, not a reference, which triggered assertion.

Unfortunately, I don't know why it started happening only with 1.14/1.15... Didn't find any relevant changes in the TF change logs.

Reply:
Yea, don't explain here, but in the code with a comment.
Also, what does it mean to be a reference or a value?